action/protocol/staking/candidate_statemanager.go

```diff
@@ -30,6 +29,9 @@ type (
 		GetBySelfStakingIndex(uint64) *Candidate
 		Upsert(*Candidate) error
 		Commit() error
+		// bucket-related
+		protocol.StateReadWriter
+		Load(string, interface{}) error
```

Comment:
if this is the purpose you create new interface, I think you are not on the right track. Our goal is to remove protocol.StateXXX from this interface definition.

Reply:
There's some interfaces cannot be removed in protocol.StateReadWriter,like State,PutState,DelState,used in handleXXX